### PR TITLE
Receive ARP as a parameter

### PR DESCRIPTION
**Pull request by dleitee**

### CODE REVIEW

File: packages/analytics/src/pingback.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ function sendPingbacks() {
 
 const debouncedPingbackEvent = debounce(1000, sendPingbacks)
 
-const pingback = ({ gif, userId, eventType, actionType, attributes, queueEvents = true }: Pingback) => {
+const pingback = ({ gif, userId, eventType, actionType, attributes, queueEvents = true, arp }: Pingback) => {
```
Comment (giannif) [inline packages/analytics/src/pingback.ts:21]: @dleitee maybe we rename this param to `analyticsResponsePayload`, it's for the `Pingback` type, and we leave `analytics_response_payload` of course for `PingbackEvent`, which has a bunch of underscore props already

File: packages/analytics/src/pingback.ts [hunk 2]
```diff
@@ -27,20 +34,16 @@ const pingback = ({ gif, userId, eventType, actionType, attributes, queueEvents
...
-            Logger.debug(`no pingback for ${gif.id}, not all endpoints have ARPs`)
-            // abort pingback, analytics_response_payload is required for gif events
-            return
-        }
-        const gifEvent = newEvent as PingbackGifEvent
-        gifEvent.analytics_response_payload = `${gif.analytics_response_payload}${
+    // add verification mode
+    if (newEvent.analytics_response_payload) {
+        newEvent.analytics_response_payload = `${newEvent.analytics_response_payload}${
             Logger.ENABLED ? '&mode=verification' : ''
```
Comment (giannif) [inline packages/analytics/src/pingback.ts:47]: Maybe we can remove `&mode=verification`? Didn't we enable that on all pingbacks?

Comment (dleitee) [reply packages/analytics/src/pingback.ts:47]: We did, but they might disable it soon

Comment (giannif) [reply packages/analytics/src/pingback.ts:47]: @dleitee cool, let's leave it then 👍

File: packages/react-components/src/util/pingback.ts [hunk 3]
```diff
@@ -9,7 +9,7 @@ const firePingback = (actionType: PingbackActionType) => (
     attributes: PingbackAttributes = {}
 ) =>
     pingback({
-        gif,
+        analyticsResponsePayload: gif.analytics_response_payload,
```
Comment (giannif) [inline packages/react-components/src/util/pingback.ts:12]: @dleitee sometimes gifs don't have an `analytics_response_payload`, even though I typed it as always there (maybe we should change that?). 

Regardless, here I think we should not fire this if `gif.analytics_response_payload` doesn't exist. 

```typescript 
const firePingback = (actionType: PingbackActionType) => (
    gif: IGif,
    userId: string | number | undefined,
    target: HTMLElement,
    attributes: PingbackAttributes = {}
) =>
    gif.analytics_response_payload && pingback({
        analyticsResponsePayload: gif.analytics_response_payload,
        userId,
        actionType,
        attributes: { position: JSON.stringify(getClientRect(target)), ...attributes },
    })

Comment (dleitee) [reply packages/react-components/src/util/pingback.ts:12]: ok, not a problem ;D

**Review by giannif — APPROVED** [review]

**Pull request merged**